### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by thomasdigby**
#### What have you done
Added some tests to cover server responses and output HTML. You can check the full list of tests on this [Trello ticket](https://trello.com/c/BdgnJwgP/48-integration-testing)

#### Why have you done it
Tests were a requirement before we started any project that utilised Tapestry.

Feedback please 🍭

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -56,6 +57,7 @@ export default class Tapestry {
     this.server.start(err => {
       if (err) error(err)
       success(`Server ready: ${this.server.info.uri}`)
+      if (typeof this.cb === 'function') this.cb()
```
Comment (jonsherrard) [inline src/server.js]: If `new Tapestry` returns a server object, we should have access to all of Hapi's server events: https://hapijs.com/api/8.8.1#server-events

```js
const server = new Tapestry(config)

server.on('start', new Build({cwd, config}))
server.on('stop', exec('cowsay "Server stopped!"'))
```

And then we could do something lik

File: test/tests/document.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    tapestry = bootServer(config, done)
+  })
+  afterEach(() => tapestry.stop())
+
+
+  it('Data is correctly loaded on page', (done) => {
+    console.log(tapestry.server.info.uri)
+    request
+      .get(tapestry.server.info.uri, (err, res, body) => {
+        expect(body).to.contain(`window.__ASYNC_PROPS__ = [{"resp":${JSON.stringify(pageData)}}]`)
```
Comment (jonsherrard) [inline test/tests/document.test.js]: Never occurred to me how awesome isomorphism/universalism can be for testing as well.

Comment (thomasdigby) [inline test/tests/document.test.js]: 👌

**Comment by thomasdigby** [comment]
[Integration Testing](https://trello.com/c/BdgnJwgP/48-integration-testing)

**Pull request merged**